conda/core/repodata.py

```diff
@@ -89,6 +89,8 @@ def query(self, package_ref_or_match_spec):
         if not self._loaded:
             self.load()
         param = package_ref_or_match_spec
+        if isinstance(param, string_types):
+            param = MatchSpec(param)
```

Comment:
This change and the other like it above just keep consistency for the three `query()` methods.